ruby/iteration.rb

```diff
@@ -19,23 +19,120 @@ def list_music
 p music_genres_capitalized
 
 #.map with an array
-cities = ["austin", "chicago", "nashville", "dallas", "houston"]
+cities = ["austin", "chicago", "nashville", "dallas", "houston", "asheville"]
 cities.map! do |city| 
 	city.capitalize
 end
 p cities
 
 #.each with a Hash
-cities_and_states = {
+	cities_and_states = {
 	"austin" => "texas",
 	"chicago" => "illinois",
 	"nashville" => "tennessee",
 	"dallas" => "texas",
-	"houston" => "texas"
+	"houston" => "texas",
+	"asheville" => "north carolina"
 }
 cities_and_states.each do |city, state|
 	puts "#{city} is in #{state}"
+	#if in symbol will need to be put in string to 
 end
+#1.  method that iterates through items , deleting
+#----------------ARRAY----------------------#
+a = [1,2,3,4,5,6,7]
+#1 method that deletes items less than 5 
+
+a.delete_if { |i| i < 5 }
+
+#----------------HASH-----------------------#
+#loop through the cities to check the state
+# for each city check the state, for each key check it's value , if value == texas, then delete
+
+def delete_texas(places_hash)
+	places_hash.delete_if {|city, state| state == "texas"
+	}
+end
+delete_texas(cities_and_states)
+p cities_and_states
+
+#now the hash has no texas cities
+
+#2. method filters a data structure for only items that do satisfy a certain condition
+#2 ----------------ARRAY----------------------#
+
+#2 method that keeps numbers less than 5
+
+def  keep(a)
+	a.select { |i| i < 5 }
+end
+#2 ----------------HASH----------------------#
+# will only work if destructive above wasn't performed
+#how do I keep the original cities_and_states for having this method placed here in the code , and where would the destructive need to go? after this method?
```

Comment:
You can comment out the other destructive method. The point was to manipulate a data structure in a couple of different ways, the methods don't build on each other.